### ISSUE TITLE
build fails on macos 11.4, nodejs 16.3.0, 

### ISSUE THREAD

**Issue by mchorvat** [PeculiarVentures/node-webcrypto-ossl#172]
hey there, 

npm package doesn't seem to install on macos. also tried to clone git repo and install to no avail. any help would be very much appreciated.

logs:
`Megan:Code marek$ git clone https://github.com/PeculiarVentures/node-webcrypto-ossl
Cloning into 'node-webcrypto-ossl'...
remote: Enumerating objects: 2800, done.
remote: Counting objects: 100% (12/12), done.
remote: Compressing objects: 100% (12/12), done.
remote: Total 2800 (delta 5), reused 0 (delta 0), pack-reused 2788
Receiving objects: 100% (2800/2800), 897.83 KiB | 2.91 MiB/s, done.
Resolving deltas: 100% (1882/1882), done.
Megan:Code marek$ cd node-webcrypto-ossl/
Megan:node-webcrypto-ossl marek$ ll
total 296
-rw-r--r--   1 marek  staff    1090 Jun  7 13:50 LICENSE
-rw-r--r--   1 marek  staff    7675 Jun  7 13:50 README.md
-rw-r--r--   1 marek  staff    4044 Jun  7 13:50 binding.gyp
-rw-r--r--   1 marek  staff     475 Jun  7 13:50 index.d.ts
drwxr-xr-x   9 marek  staff     288 Jun  7 13:50 lib
-rw-r--r--   1 marek  staff  112332 Jun  7 13:50 package-lock.json
-rw-r--r--   1 marek  staff    2292 Jun  7 13:50 package.json
-rw-r--r--   1 marek  staff     983 Jun  7 13:50 rollup.config.js
drwxr-xr-x  10 marek  staff     320 Jun  7 13:50 src
drwxr-xr-x   4 marek  staff     128 Jun  7 13:50 test
-rw-r--r--   1 marek  staff     471 Jun  7 13:50 tsconfig.json
-rw-r--r--   1 marek  staff     462 Jun  7 13:50 tslint.json
Megan:node-webcrypto-ossl marek$ npm install

> node-webcrypto-ossl@2.1.2 install
> node-gyp rebuild

gyp info it worked if it ends with ok
gyp info using node-gyp@7.1.2
gyp info using node@16.3.0 | darwin | arm64
gyp info find Python using Python version 3.9.5 found at "/opt/homebrew/opt/python@3.9/bin/python3.9"
(node:43531) [DEP0150] DeprecationWarning: Setting process.config is deprecated. In the future the property will be read-only.
(Use `node --trace-deprecation ...` to show where the warning was created)
gyp info spawn /opt/homebrew/opt/python@3.9/bin/python3.9
gyp info spawn args [
gyp info spawn args   '/opt/homebrew/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/marek/Documents/Code/node-webcrypto-ossl/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/opt/homebrew/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/marek/Library/Caches/node-gyp/16.3.0/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/Users/marek/Library/Caches/node-gyp/16.3.0',
gyp info spawn args   '-Dnode_gyp_dir=/opt/homebrew/lib/node_modules/npm/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/Users/marek/Library/Caches/node-gyp/16.3.0/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/Users/marek/Documents/Code/node-webcrypto-ossl',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
  CXX(target) Release/obj.target/nodessl/src/main.o
Apple clang version 12.0.5 (clang-1205.0.22.9)
Target: aarch64-apple-darwin20.5.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
 "/Library/Developer/CommandLineTools/usr/bin/clang" -cc1 -triple arm64-apple-macosx10.7.0 -Wdeprecated-objc-isa-usage -Werror=deprecated-objc-isa-usage -Werror=implicit-function-declaration -emit-obj -disable-free -disable-llvm-verifier -discard-value-names -main-file-name main.cpp -mrelocation-model pic -pic-level 2 -mframe-pointer=non-leaf -fno-strict-return -fno-rounding-math -munwind-tables -faligned-alloc-unavailable -target-sdk-version=11.3 -fvisibility-inlines-hidden-static-local-var -target-cpu apple-a12 -target-feature +v8.3a -target-feature +fp-armv8 -target-feature +neon -target-feature +crc -target-feature +crypto -target-feature +fullfp16 -target-feature +ras -target-feature +lse -target-feature +rdm -target-feature +rcpc -target-feature +zcm -target-feature +zcz -target-feature +sha2 -target-feature +aes -target-abi darwinpcs -fallow-half-arguments-and-returns -debug-info-kind=standalone -dwarf-version=2 -debugger-tuning=lldb -target-linker-version 650.9 -v -resource-dir /Library/Developer/CommandLineTools/usr/lib/clang/12.0.5 -dependency-file ./Release/.deps/Release/obj.target/nodessl/src/main.o.d.raw -skip-unused-modulemap-deps -MT Release/obj.target/nodessl/src/main.o -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -D NODE_GYP_MODULE_NAME=nodessl -D USING_UV_SHARED=1 -D USING_V8_SHARED=1 -D V8_DEPRECATION_WARNINGS=1 -D V8_DEPRECATION_WARNINGS -D V8_IMMINENT_DEPRECATION_WARNINGS -D _GLIBCXX_USE_CXX11_ABI=1 -D _DARWIN_USE_64_BIT_INODE=1 -D _LARGEFILE_SOURCE -D _FILE_OFFSET_BITS=64 -D BUILDING_NODE_EXTENSION -I /Users/marek/Library/Caches/node-gyp/16.3.0/include/node -I /Users/marek/Library/Caches/node-gyp/16.3.0/src -I /Users/marek/Library/Caches/node-gyp/16.3.0/deps/openssl/config -I /Users/marek/Library/Caches/node-gyp/16.3.0/deps/openssl/openssl/include -I /Users/marek/Library/Caches/node-gyp/16.3.0/deps/uv/include -I /Users/marek/Library/Caches/node-gyp/16.3.0/deps/zlib -I /Users/marek/Library/Caches/node-gyp/16.3.0/deps/v8/include -I ../node_modules/nan -I/usr/local/include -stdlib=libc++ -stdlib=libc++ -internal-isystem /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1 -internal-isystem /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/local/include -internal-isystem /Library/Developer/CommandLineTools/usr/lib/clang/12.0.5/include -internal-externc-isystem /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include -internal-externc-isystem /Library/Developer/CommandLineTools/usr/include -O3 -Wall -Wendif-labels -W -Wno-unused-parameter -Wno-reorder-init-list -Wno-implicit-int-float-conversion -Wno-c99-designator -Wno-final-dtor-non-final-class -Wno-extra-semi-stmt -Wno-misleading-indentation -Wno-quoted-include-in-framework-header -Wno-implicit-fallthrough -Wno-enum-enum-conversion -Wno-enum-float-conversion -Wno-elaborated-enum-base -std=c++11 -fdeprecated-macro -fdebug-compilation-dir /Users/marek/Documents/Code/node-webcrypto-ossl/build -ferror-limit 19 -stack-protector 1 -fstack-check -mdarwin-stkchk-strong-link -fblocks -fencode-extended-block-signature -fno-rtti -fregister-global-dtors-with-atexit -fgnuc-version=4.2.1 -fcxx-exceptions -fexceptions -fmax-type-align=16 -fcommon -fcolor-diagnostics -vectorize-loops -vectorize-slp -clang-vendor-feature=+disableNonDependentMemberExprInCurrentInstantiation -fno-odr-hash-protocols -mllvm -disable-aligned-alloc-awareness=1 -o Release/obj.target/nodessl/src/main.o -x c++ ../src/main.cpp
clang -cc1 version 12.0.5 (clang-1205.0.22.9) default target arm64-apple-darwin20.5.0
ignoring nonexistent directory "/Users/marek/Library/Caches/node-gyp/16.3.0/src"
ignoring nonexistent directory "/Users/marek/Library/Caches/node-gyp/16.3.0/deps/openssl/config"
ignoring nonexistent directory "/Users/marek/Library/Caches/node-gyp/16.3.0/deps/openssl/openssl/include"
ignoring nonexistent directory "/Users/marek/Library/Caches/node-gyp/16.3.0/deps/uv/include"
ignoring nonexistent directory "/Users/marek/Library/Caches/node-gyp/16.3.0/deps/zlib"
ignoring nonexistent directory "/Users/marek/Library/Caches/node-gyp/16.3.0/deps/v8/include"
ignoring nonexistent directory "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/local/include"
ignoring nonexistent directory "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/Library/Frameworks"
#include "..." search starts here:
#include <...> search starts here:
 /Users/marek/Library/Caches/node-gyp/16.3.0/include/node
 ../node_modules/nan
 /usr/local/include
 /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1
 /Library/Developer/CommandLineTools/usr/lib/clang/12.0.5/include
 /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include
 /Library/Developer/CommandLineTools/usr/include
 /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks (framework directory)
End of search list.
In file included from ../src/main.cpp:1:
In file included from ../src/node/common.h:4:
In file included from ../src/node/../core/common.h:8:
In file included from ../node_modules/nan/nan.h:56:
In file included from /Users/marek/Library/Caches/node-gyp/16.3.0/include/node/node.h:63:
In file included from /Users/marek/Library/Caches/node-gyp/16.3.0/include/node/v8.h:30:
/Users/marek/Library/Caches/node-gyp/16.3.0/include/node/v8-internal.h:452:38: error: no template named 'remove_cv_t' in namespace 'std'; did you mean 'remove_cv'?
            !std::is_same<Data, std::remove_cv_t<T>>::value>::Perform(data);
                                ~~~~~^~~~~~~~~~~
                                     remove_cv
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/type_traits:776:50: note: 'remove_cv' declared here
template <class _Tp> struct _LIBCPP_TEMPLATE_VIS remove_cv
                                                 ^
1 error generated.
make: *** [Release/obj.target/nodessl/src/main.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/opt/homebrew/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (node:events:394:28)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:290:12)
gyp ERR! System Darwin 20.5.0
gyp ERR! command "/opt/homebrew/Cellar/node/16.3.0/bin/node" "/opt/homebrew/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/marek/Documents/Code/node-webcrypto-ossl
gyp ERR! node -v v16.3.0
gyp ERR! node-gyp -v v7.1.2
gyp ERR! not ok 
npm ERR! code 1
npm ERR! path /Users/marek/Documents/Code/node-webcrypto-ossl
npm ERR! command failed
npm ERR! command sh -c node-gyp rebuild

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/marek/.npm/_logs/2021-06-07T11_50_22_267Z-debug.log
Megan:node-webcrypto-ossl marek$ sw_vers
ProductName:	macOS
ProductVersion:	11.4
BuildVersion:	20F71
Megan:node-webcrypto-ossl marek$ `

**Comment by microshine** [comment]
I'll check it out. 

Have you seen (@peculiar/webcrypto)[https://www.npmjs.com/package/@peculiar/webcrypto]? It uses NodeJS native `crypto` module and doesn't require C++ plugin

**Comment by mchorvat** [comment]
on a second thought... might be issue with node-gyp though... not sure :(
thanks for pointing out webcrypto, but since the ultimate goal is to get fido2-library running (https://www.npmjs.com/package/fido2-library) and it is dependent on node-webcrypto-ossl, i might need to get this one working :)

**Comment by microshine** [comment]
The issue is fixed. I published the new version `node-webcrypto-ossl@2.1.3`
